src/main/java/com/networknt/schema/ContainsValidator.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2016 Network New Technologies Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.networknt.schema;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.Set;
+
+public class ContainsValidator extends BaseJsonValidator implements JsonValidator {
+    private static final Logger logger = LoggerFactory.getLogger(ContainsValidator.class);
+
+    private JsonSchema schema;
+
+    public ContainsValidator(String schemaPath, JsonNode schemaNode, JsonSchema parentSchema, ValidationContext validationContext) {
+        super(schemaPath, schemaNode, parentSchema, ValidatorTypeCode.CONTAINS, validationContext);
+        if (schemaNode.isObject() || schemaNode.isBoolean()) {
+            schema = new JsonSchema(validationContext, getValidatorType().getValue(), parentSchema.getCurrentUri(), schemaNode, parentSchema);
+        }
+
+        parseErrorCode(getValidatorType().getErrorCodeKey());
+    }
+
+    public Set<ValidationMessage> validate(JsonNode node, JsonNode rootNode, String at) {
+        debug(logger, node, rootNode, at);
+
+
+        if (!node.isArray()) {
+            // ignores non-arrays
+            return Collections.emptySet();
+        }
+
+        if (node.isEmpty()) {
+            // Array was empty
+            return buildErrorMessageSet(at);
+        } else if (node.isArray()) {
+            boolean contains = false;
+            int i = 0;
+            for (JsonNode n : node) {
+                contains = contains || schema.validate(n, rootNode, at + "[" + i + "]").isEmpty();
+                i++;
```

Comment:
Perhaps adding a `if(contains) break;` would be ideal, to decrease interations.

Reply:
For most of the scenarios, it doesn't matter as the number of elements won't be big. However, we are trying to provide a generic library and we cannot assume the use cases from users. There might be some crazy use cases with too many elements so I think you are right about to add the break.

Reply:
I have pushed a new version, where I made it short circuit on the first match. This also somewhat simplifies the code.